rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -2124,3 +2492,101 @@ func Create_TradeState: <"Creates a single trade state by applying primitive ins
                 activityDate: empty, // TODO: set this property
                 ...
             }
+
+func ExtractBeforeEconomicTerms:
+    inputs:
+        businessEvent BusinessEvent (1..1)
+    output:
+        economicTerms EconomicTerms (0..1)
+
+    set economicTerms:
+        businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+
+func ExtractAfterEconomicTerms:
+    inputs:
+        businessEvent BusinessEvent (1..1)
+    output:
+        economicTerms EconomicTerms (0..1)
+
+    set economicTerms:
+        FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+
+
+func ExtractOpenEconomicTerms:
+    inputs:
+        businessEvent BusinessEvent (1..1)
+    output:
+        economicTerms EconomicTerms (0..1)
+
+    set economicTerms:
+        FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+
+func ExtractBeforeTradableProduct:
+    inputs:
+        businessEvent BusinessEvent (1..1)
+    output:
+        tradableProduct TradableProduct (0..1)
+
+    set tradableProduct:
+        businessEvent -> instruction only-element -> before -> trade -> tradableProduct
+
+func ExtractAfterTradableProduct:
+    inputs:
+        businessEvent BusinessEvent (1..1)
+    output:
+        tradableProduct TradableProduct (0..1)
+
+    set tradableProduct:
+        FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct
+
+func ExtractBeforeTradePurchasePrice:
+    inputs:
+        beforeTradableProduct TradableProduct (1..1)
+    output:
+        value number (0..*)
+
+    set value:
+        beforeTradableProduct -> tradeLot only-element -> priceQuantity
+        extract [ item -> price ]
+        then flatten
+        then filter [ item -> priceType = PriceTypeEnum -> InterestRate ]
+        then extract [ item -> value ]
```

Comment:
This extracts the interest rate rather than the "purchase price". The purchase price is the `quantity` that is associated with that interest rate rather than the `price`.

Reply:
Fixed.